frontend/components.js

```diff
@@ -262,37 +305,46 @@ class SendAdaClass extends Component {
             placeholder: 'Receiving address',
             size: '28',
             value: sendAddress,
-            onInput: inputAddress,
+            onInput: updateAddress,
+            autocomplete: 'nope',
           })
         ),
-        h(
-          'div',
-          {class: 'amount-label-row'},
-          h('label', undefined, h('span', undefined, 'Amount')),
-          h('span', {class: 'transaction-fee'}, `+ ${transactionFee} transaction fee`)
-        ),
+        (transactionFee > 0) ?
+          h(
+            'div',
+            {class: 'amount-label-row'},
+            h('label', undefined, h('span', undefined, 'Amount')),
+            h('span', {class: 'transaction-fee'}, `+ ${transactionFee} transaction fee`)
+          ) : '',
+        sendAmountValidationMsg ? h('p', {class: 'alert'}, sendAmountValidationMsg) : '',
         h(
           'div',
           {class: 'webflow-style-input send-input'},
           h('input', {
-            type: 'text',
+            type: 'number',
+            min: 0,
+            max: balance,
             id: 'send-amount',
             name: 'send-amount',
             placeholder: 'Amount',
             size: '28',
-            value: sendAmountFieldValue,
-            onInput: inputAmount,
+            value: sendAmount,
+            onInput: updateAmount,
+            autocomplete: 'nope',
           }),
-          h('span', {style: `color: ${feeRecalculating ? 'red' : 'green'}`}, `= ${totalAmount} ADA`)
+          (transactionFee > 0) ?
+            h('span', {style: `color: ${feeRecalculating ? 'red' : 'green'}`}, `= ${totalAmount} ADA`)
+            : '',
         ),
-        feeRecalculating
+        (feeRecalculating || sendAmount !== feeCalculatedFrom.amount || sendAddress !== feeCalculatedFrom.address)
```

Comment:
Why so many conditions? Isn't the feeRecalculating variable enough? I tried removing the remaining conditions and the validation seems to work OK. Are there any corner cases I'm not aware of?